webpack/components/extensions/Hosts/BulkActions/BulkChangeHostCollectionsModal/BulkChangeHostCollectionsModal.js

```diff
@@ -0,0 +1,323 @@
+
+  const selectionToolbar = (
+    <ToolbarItem key="selectAll">
+      <SelectAllCheckbox
+        {...{
+          selectAll: noop,
+          selectOne,
+          updateSearchQuery,
+          selectNone,
+          selectPage,
+          selectedCount,
+          pageRowCount: pageStats.pageRowCount,
+          areAllRowsSelected,
+          areAllRowsOnPageSelected,
+        }}
+        totalCount={total}
+        areAllRowsOnPageSelected={areAllRowsOnPageSelected()}
+        areAllRowsSelected={areAllRowsSelected()}
+      />
+    </ToolbarItem>
+  );
+
+  const handleModalClose = () => {
+    setHCIds([]);
+    setAddRadioChecked(true);
+    selectNone();
+    closeModal();
+  };
+
+  const handleError = (error) => {
+    const message = error?.response?.data?.error?.message || __('An error occurred while updating host collections');
+    dispatch(addToast({ type: 'danger', message }));
+  };
+
+  const handleSuccess = (response) => {
+    const messages = response?.data?.displayMessages;
+    const message = Array.isArray(messages) && messages.length > 0
+      ? (
+        <>
+          {messages.map((msg, index) => (
+            <div key={`host-status-${index}`}>{msg}</div>
+          ))}
+        </>
+      )
+      : __('Host collections updated successfully');
+
+    dispatch(addToast({ type: 'success', message }));
+    dispatch(APIActions.get({
+      key: API_REQUEST_KEY,
+      url: foremanUrl(HOSTS_API_PATH),
+    }));
+    handleModalClose();
+  };
+
+  const handleSave = () => {
+    const requestBody = {
+      included: {
+        search: fetchBulkParams(),
+      },
+      host_collection_ids: HCIds,
+      organization_id: orgId,
+    };
+
+    if (addRadioChecked) {
+      dispatch(bulkAddHostCollections(requestBody, handleSuccess, handleError));
+    } else {
+      dispatch(bulkRemoveHostCollections(requestBody, handleSuccess, handleError));
+    }
+  };
+
+  const isLoadingHostCollections = hostCollectionsLoadingStatus === STATUS.PENDING;
+  const isSaveDisabled = HCIds.length === 0 || isLoadingHostCollections;
+
+  const modalActions = [
+    <Button
+      key="save"
+      ouiaId="bulk-change-host-collections-modal-save-button"
+      variant="primary"
+      onClick={handleSave}
+      isDisabled={isSaveDisabled}
+      isLoading={isLoadingHostCollections}
+    >
+      {__('Save')}
+    </Button>,
+    <Button
+      key="cancel"
+      ouiaId="bulk-change-host-collections-modal-cancel-button"
+      variant="link"
+      onClick={handleModalClose}
+    >
+      {__('Cancel')}
+    </Button>,
+  ];
+
+  const columns = {
+    name: {
+      title: __('Host collection'),
+      wrapper: hc => hc?.name || '',
+      isSorted: true,
+      weight: 50,
+    },
+    limit: {
+      title: __('Limit'),
+      wrapper: (hc) => {
+        const totalLimit = hc?.unlimited_hosts ? 'unlimited' : (hc?.max_hosts || 0);
+        const count = hc?.total_hosts || 0;
+        return `${count}/${totalLimit}`;
```

Comment:
In addition to my comment above (or maybe instead of it?) would be nice to add a warning icon here if the host limit is met or exceeded.